### PR TITLE
Update CMake for dynamic linking - Add test build in Github actions

### PR DESCRIPTION
**Pull request by OdyX**
:wave: @Skywalker13 . This is my work-in-progress to address #8 against master.

Comments welcome!

### CODE REVIEW

**Comment by OdyX** [comment]
For the record, I can confirm that with these changes, a dynamically-linked Debian packaging build works; sounds and animations included! :sunglasses: .

https://salsa.debian.org/games-team/blupimania/-/pipelines/613595

File: CMakeLists.txt [hunk 1]
```diff
@@ -220,9 +217,9 @@ if ("${STATIC_BUILD}")
   endif ()
 else ()
   set (blupimania_DEPS
-    ${SDL2_LIBRARIES}
-    ${SDL2_MIXER_LIBRARIES}
-    ${SDL2_IMAGE_LIBRARIES}
+    SDL2::SDL2
+    SDL2_image::SDL2_image
+    SDL2_mixer::SDL2_mixer
     ${CURL_LIBRARIES}
```
Comment (Skywalker13) [inline CMakeLists.txt:225]: IIRC I had problems with this way on macOS or Windows/MSYS2.. But I must check again because I'm not sure about that.

Comment (Skywalker13) [reply CMakeLists.txt:225]: I tested against all operating systems and it works fine

**Comment by Skywalker13** [comment]
It's fine for me, I merge your changes

Thanks

**Pull request merged**